tests/histogram.rs

```diff
@@ -395,3 +398,50 @@ fn scaled_set_to() {
     h2.set_to(&h1).unwrap();
     are_equal(&h1, &h2);
 }
+
+
+#[test]
+fn random_write_full_value_range_precision_5_no_panic() {
+    let mut h = Histogram::<u64>::new_with_bounds(1, u64::max_value(), 5).unwrap();
+
+    let mut rng = rand::weak_rng();
+
+    for _ in 0..1_000_000 {
+        let mut r: u64 = rng.gen();
+        if r == 0 {
+            r = 1;
+        }
+
+        h.record(r).unwrap();
```

Comment:
I think that syntax is unintuitive. Bias warning: I'm generally anti-operator-overloading because it lets everyone come up with their own mini-language, and that hasn't worked out well in C++. (You can end up so far in your own world of operator overloading that you end up having a hard time teaching a new team member how to write your custom variant of C++, etc.) No doubt because of that bias, it didn't even occur to me to use `+=`. To me, `+=` is a thing I can't trivially navigate to find the source of, and  unlike methods (in Rust anyway) it's also overloaded in the type sense: the same syntax can be used to add a histogram into the receiver of the method call, which makes it less comprehensible to my eyes.